src/app/zap-templates/templates/app/helper.js

```diff
@@ -423,6 +388,89 @@ function asTypedLiteral(value, type)
   return templateUtil.templatePromise(this.global, promise)
 }
 
+async function asTypedExpression(value, type)
+{
+  const valueIsANumber = !isNaN(value);
+  if (!value || valueIsANumber) {
+    return asTypedLiteral.call(this, value, type);
+  }
+
+  const tokens = value.split(' ');
+  if (tokens.length < 2) {
+    return asTypedLiteral.call(this, value, type);
+  }
+
+  const hasVariable = tokens.find(token => {
+    const isOperator = [ '+', '-', '/', '*', '%' ].indexOf(token) != -1;
+    if (isOperator) {
+      return false;
+    }
+
+    const isNumber = !isNaN(token.replace(/ULL$|UL$|U$|LL$|L$/i, ''));
+    if (isNumber) {
+      return false;
+    }
+
+    return true;
+  });
+
+  if (hasVariable) {
+    let isNullable = false;
+    if (this.parent.isAttribute) {
+      isNullable = this.parent.attributeObject.isNullable;
+    } else if (this.parent.isCommand) {
+      isNullable = this.parent.commandObject.isNullable;
+    } else if (this.parent.isEvent) {
+      isNullable = this.parent.eventObject.isNullable;
```

Comment:
These don't make sense to me.  Commands and events are not nullable; their fields are.

Shouldn't we be examining `this.isNullable` in general, or something?